src/main/java/edu/ucdavis/dss/ipa/services/jpa/JpaUserRoleService.java

```diff
@@ -341,5 +344,60 @@ public Instructor findOrAddActiveInstructor(Workgroup workgroup, User user) {
 
 		return uniqueInstructors;
 	}
+
+
+	/**
+	 * Returns a list of SupportStaff entities that are tied to userRoles for masters/phd/instructionalSupport roles.
+	 * These 3 populations will all have instructionalSupportStaff entities, and represent people
+	 * an academic planner might want to assign to an instructionalSupportAssignment.
+	 * @param workgroupId
+	 * @return
+	 */
+	@Override
+	public List<SupportStaff> findActiveSupportStaffByWorkgroupId(long workgroupId) {
+		List<String> roleTokens = new ArrayList<>(Arrays.asList("studentMasters", "studentPhd", "instructionalSupport"));
+		List<UserRole> activeUserRoles = this.findByWorkgroupIdAndRoleTokens(workgroupId, roleTokens);
+
+		List<SupportStaff> activeSupportStaffList = new ArrayList<SupportStaff>();
+
+		for (UserRole userRole : activeUserRoles) {
+			SupportStaff supportStaff = supportStaffService.findOrCreate(userRole.getUser().getFirstName(), userRole.getUser().getLastName(), userRole.getUser().getEmail(), userRole.getUser().getLoginId());
+			activeSupportStaffList.add(supportStaff);
+		}
+
+		return activeSupportStaffList;
+	}
+
+	private List<UserRole> findByWorkgroupIdAndRoleTokens(long workgroupId, List<String> roleTokens) {
```

Comment:
You should be querying the database with a list of tokens, not merely extracting the old code out. Think about performance: one query is usually better than multiple queries, especially if the data is small.